src/service/action-impl.js

```diff
@@ -871,7 +872,9 @@ export class ActionService {
     }
 
     if (Object.keys(detail).length > 0) {
-      event.detail = detail;
+      try {
+        event.detail = detail;
+      } catch (e) {}
```

Comment:
Nit: We can use a [parameter-less catch clause](https://v8.dev/features/optional-catch-binding).

```suggestion
      } catch {}
```